apps/asap-server/test/controllers/teams.test.ts

```diff
@@ -303,51 +303,90 @@ describe('Team controller', () => {
       });
     });
 
-    test('Should parse the team user correctly when the avatar is null', async () => {
-      const teamId = 'team-id-1';
-      const user = fetchByIdUserResponse.items[0];
-      user.data.avatar.iv = null as any;
-
-      const userResponse = {
-        total: 1,
-        items: [user],
+    describe('Avatar', () => {
+      const user = {
+        ...fetchByIdUserResponse.items[0],
+        data: {
+          ...fetchByIdUserResponse.items[0].data,
+          avatar: { iv: 'test avatar' },
+        },
       };
 
-      nock(config.baseUrl)
-        .post(`/api/content/${config.appName}/graphql`, {
-          query: buildGraphQLQueryFetchTeam(teamId),
-        })
-        .reply(200, graphQlTeamResponse)
-        .get(`/api/content/${config.appName}/users`)
-        .query({
-          $filter: `data/teams/iv/id eq '${teamId}'`,
-        })
-        .reply(200, userResponse);
+      test('Should parse the team user correctly when the avatar is null', async () => {
+        const teamId = 'team-id-1';
+
+        user.data.avatar.iv = null as any;
+
+        const userResponse = {
+          total: 1,
+          items: [user],
+        };
+
+        nock(config.baseUrl)
+          .post(`/api/content/${config.appName}/graphql`, {
+            query: buildGraphQLQueryFetchTeam(teamId),
+          })
+          .reply(200, graphQlTeamResponse)
+          .get(`/api/content/${config.appName}/users`)
+          .query({
+            $filter: `data/teams/iv/id eq '${teamId}'`,
+          })
+          .reply(200, userResponse);
+
+        const result = await teams.fetchById(teamId, mockUser);
+
+        const expectedResponse = {
+          ...fetchTeamByIdExpectation,
+          members: [
+            {
+              ...fetchTeamByIdExpectation.members[0],
+              avatarUrl: undefined,
+            },
+          ],
+        };
 
-      const result = await teams.fetchById(teamId, mockUser);
+        expect(result).toEqual(expectedResponse);
+      });
 
-      const expectedResponse = {
-        ...fetchTeamByIdExpectation,
-        members: [
-          {
-            ...fetchTeamByIdExpectation.members[0],
-            avatarUrl: undefined,
-          },
-        ],
-      };
+      test('Should parse the team user correctly when the avatar is undefined', async () => {
+        const teamId = 'team-id-1';
+
+        delete (user.data as any).avatar;
```

Comment:
Why do we need casting here?

Reply:
well in this case the type of `user` is actually being inferred from an object here:

https://github.com/yldio/asap-hub/pull/840/files/238050457b37380277be8eb4524699bd66498447#diff-79015ebee0b55ad2336fccbd627b61384c730904ec40c7cf86a3ad57c904d90eR307

however - even with extra typings to make the `user` become `RestUser` the `avatar` is still marked as required and so we need to cast it as `any` before deleting the property. 

We do know the types are invalid for squidex lists and we have a ticket to address it here:

https://trello.com/c/oJqbA3z9/1106-update-typings-to-accommodate-for-iv-null-squidex-response

Reply:
Alright! 👍🏽